test/conftest.py

```diff
@@ -202,33 +257,52 @@ def image_url():
     return f'file://{path}'
 
 
-def setup_db(db, **kwargs):
+def create_db(CFG, **kwargs):
     # TODO: support more parameters to control the setup
+    db = build_datalayer(CFG)
+    if kwargs.get('empty', False):
+        return db
+
     add_encoders(db)
-    n_data = kwargs.get('n_data', GLOBAL_TEST_N_DATA_POINTS)
-    add_random_data(db, number_data_points=n_data)
+    n_data = kwargs.get('n_data', LOCAL_TEST_N_DATA_POINTS)
+
+    # prepare data
+    is_mongodb_bachend = isinstance(db.databackend, MongoDataBackend)
+    if is_mongodb_bachend:
+        add_random_data_to_mongo_db(db, number_data_points=n_data)
+    else:
+        add_random_data_to_sql_db(db, number_data_points=n_data)
+
+    # prepare models
     if kwargs.get('add_models', True):
         add_models(db)
+
+    # prepare vector index
     if kwargs.get('add_vector_index', True):
         add_vector_index(db)
 
-
-@pytest.fixture(scope='session')
-def db() -> Datalayer:
-    db = build_datalayer(CFG, data_backend=MONGOMOCK_URI)
-    setup_db(db)
     return db
 
 
 @pytest.fixture
-def local_db(request) -> Datalayer:
-    db = build_datalayer(CFG, data_backend=MONGOMOCK_URI)
-    setup_config = getattr(request, 'param', {'n_data': LOCAL_TEST_N_DATA_POINTS})
-    setup_db(db, **setup_config)
-    return db
+def db(request, monkeypatch) -> Iterator[Datalayer]:
+    # TODO: Use pre-defined config instead of dict here
+    db_type, setup_config = (
+        request.param if hasattr(request, 'param') else ("mongodb", None)
+    )
+    setup_config = setup_config or {}
+    if db_type == "mongodb":
+        monkeypatch.setattr(CFG, 'data_backend', MONGOMOCK_URI)
+    elif db_type == "sqldb":
+        monkeypatch.setattr(CFG, 'data_backend', SQLITE_URI)
 
+    db = create_db(CFG, **setup_config)
+    yield db
 
-@pytest.fixture
-def local_empty_db(request) -> Datalayer:
-    db = build_datalayer(CFG, data_backend=MONGOMOCK_URI)
-    return db
+    if db_type == "mongodb":
+        db.drop(force=True)
+    elif db_type == "sqldb":
+        db.artifact_store.drop(force=True)
+        tables = db.databackend.conn.list_tables()
+        for table in tables:
+            db.databackend.conn.drop_table(table, force=True)
```

Comment:
Standardize fixture creation using configuration settings, with support for MongoDB and SQL databases.

Reply:
# TODO: Use pre-defined config instead of dict here

Reply:
This is an optimization I found while working, which involves combining the MongoDB and SQL DB fixtures. What do you think, should we do this? Or should we keep the fixtures separate? If we separate them, it might be a bit more troublesome to use different fixtures for a single test case.

WDYT? @thejumpman2323 @blythed